extmod/mpbthci.h

```diff
@@ -47,6 +49,7 @@ int mp_bluetooth_hci_uart_deinit(void);
 int mp_bluetooth_hci_uart_set_baudrate(uint32_t baudrate);
 int mp_bluetooth_hci_uart_any(void);
 int mp_bluetooth_hci_uart_readchar(void);
+int mp_bluetooth_hci_uart_readpacket(mp_bluetooth_hci_uart_readchar_t handler);
```

Comment:
Because a given transport can only implement either one of these, we should make it a #define in `mpbthci.h`

```
#define MICROPY_PY_BLUETOOTH_HCI_READ_MODE_BYTE = 0
#define MICROPY_PY_BLUETOOTH_HCI_READ_MODE_PACKET = 1

#ifndef MICROPY_PY_BLUETOOTH_HCI_READ_MODE
#define MICROPY_PY_BLUETOOTH_HCI_READ_MODE MICROPY_PY_BLUETOOTH_HCI_READ_MODE_BYTE
#endif
```

Then we don't need to make `mp_bluetooth_hci_uart_readchar` and `mp_bluetooth_hci_uart_readpacket` weak, and this will force one of them to be properly defined.